src/ng/compile.js

```diff
@@ -1877,6 +1877,36 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
         return value || null;
       }
 
+      function setupControllers($element, attrs, transcludeFn, controllerDirectives, isolateScope, scope) {
+        var elementControllers = createMap();
+        for (var controllerKey in controllerDirectives) {
```

Comment:
Well, `angular.forEach` ends up using `for ... in` in any case, except that it checks the `hasOwnProperty` first.  In this case, I am pretty confident that `controllerDirectives` will not contain any dodgy properties, especially since @jbedard has used `createMap()` to ensure that it doesn't derive from `Object`.